libkbfs/folder_block_ops.go

```diff
@@ -787,8 +787,11 @@ func (fbo *folderBlockOps) GetDirtyEntry(
 func (fbo *folderBlockOps) Lookup(
 	ctx context.Context, lState *lockState, kmd KeyMetadata,
 	dir Node, name string) (Node, DirEntry, error) {
-	fbo.blockLock.RLock(lState)
-	defer fbo.blockLock.RUnlock(lState)
+	// Take the full lock, so that the blockfetcher can't r-unlock us
+	// when fetching a remote block, which could lead to a race
+	// (KBFS-1760).
+	fbo.blockLock.Lock(lState)
```

Comment:
Have you considered creating a new `blockReqType`, say, `blockLookup`, and not doing an `RUnlock` then? Would be a shame to lose parallelism here.

But I'm okay with this, too.